great_expectations/datasource/data_connector/file_path_data_connector.py

```diff
@@ -124,7 +124,12 @@ def get_batch_definition_list_from_batch_request(
             A list of BatchDefinition objects that match BatchRequest
 
         """
-        batch_request_base: BatchRequestBase = cast(BatchRequestBase, batch_request)
+        # <WILL> TODO 202103 some methods still call this method with BatchRequestBase. Refactor to only allow BatchRequest
```

Comment:
I am not sure if the `"TODO"` here is the best approach.  Personally, I would rather see `BatchRequestBase` disappear entirely.  It should not be difficult, and the code will be a lot easier to interpret.  I suggested a way to do it during my previous review of this PR; would be more than happy to discuss/brainstorm together in greater detail.  Thanks!

Reply:
Would love to discuss more in greater detail :) also wondering if they should be separate PRs.